platzky/plugin/plugin_loader.py

```
@@ -87,119 +166,68 @@ def _process_legacy_plugin(
-        PluginError: if plugin processing fails
+        PluginError: if a configured plugin cannot be loaded or initialised
     """
-    plugins_data = app.db.get_plugins_data()
-
-    for plugin_data in plugins_data:
-        plugin_config = plugin_data["config"]
-        plugin_name = plugin_data["name"]
+    try:
+        plugins_data = app.db.get_plugins_data()
+    except ValidationError as e:
+        raise PluginError(f"Invalid plugin configuration in database: {e}") from e
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)